src/daemon/rpmostreed-os.c

```diff
@@ -1699,50 +1699,44 @@ os_handle_download_deploy_rpm_diff (RPMOSTreeOS *interface,
 static gboolean
 refresh_cached_update (RpmostreedOS *self, GError **error)
 {
-  const char *name = rpmostree_os_get_name (RPMOSTREE_OS (self));
-  OstreeSysroot *sysroot = rpmostreed_sysroot_get_root (rpmostreed_sysroot_get ());
-  OstreeRepo *repo = ostree_sysroot_repo (sysroot);
-
-  /* if we're not handling the system sysroot, then let's just skip all this (e.g. `make
-   * check` tests) */
-  const char *sysroot_path = gs_file_get_path_cached (ostree_sysroot_get_path (sysroot));
-  if (!g_str_equal (sysroot_path, "/"))
-    return TRUE;
-
-  /* Note here we're *not* using rpmostree_syscore_get_origin_merge_deployment(): cached
-   * updates are always relative to the booted/merge deployment; e.g. we still want to be
-   * able to show details about a pending deployment. */
-  g_autoptr(OstreeDeployment) merge_deployment =
-    ostree_sysroot_get_merge_deployment (sysroot, name);
-
   g_autoptr(GVariant) cached_update = NULL;
-  if (merge_deployment)
+
+  gsize n;
+  g_autofree char *contents = NULL;
+  g_autoptr(GError) local_error = NULL;
+  if (!g_file_get_contents (RPMOSTREE_AUTOUPDATES_CACHE_FILE, &contents, &n, &local_error))
```

Comment:
We do have `glnx_fd_readall_bytes()` which gives you a `GBytes` which integrates more nicely with `GVariant`.